### ISSUE TITLE
false positive on save in template editor 

### ISSUE THREAD

**Issue by tux3l** [coreruleset/wordpress-rule-exclusions-plugin#44]
I reproduced this on a completely fresh installation with OWASP CRS 4.2.0 and wordpress-rule-exclusions-plugin v1.0.0 with an wordpress installation version 6.5.3 and 6.5.4 with theme Twenty Twenty-Three Version: 1.4

I have nothing done in crs-setup.conf from  nor anything changed in the wordpress-rule-exclusions-plugin. If this is my mistake please help me to understand what i missed and why. Thanks.

Error from modsec_audit.log (i hope this shows enough, otherwise i can provide more details if you tell me what is needed):

```
--18fc1d05-A--
[06/Jun/2024:11:28:08.255620 +0000] ZmGdSKiGpWWaBe02xHB0yAAAABg 1XX.YYY.ZZZ.1 60724 1XX.YYY.ZZZ.185 80
--18fc1d05-B--
POST /index.php?rest_route=%2Fwp%2Fv2%2Ftemplates%2Ftwentytwentythree%2F%2Fpage&_locale=user HTTP/1.1
Host: picard.test.domain.de
User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:126.0) Gecko/20100101 Firefox/126.0
Accept: application/json, */*;q=0.1
Accept-Language: de,en-US;q=0.7,en;q=0.3
Accept-Encoding: gzip, deflate
Referer: http://picard.test.domain.de/wp-admin/site-editor.php?postType=wp_template&postId=twentytwentythree%2F%2Fpage&canvas=edit
X-WP-Nonce: dede7178fa
X-HTTP-Method-Override: PUT
Content-Type: application/json
Content-Length: 1172
Origin: http://picard.test.domain.de
Connection: keep-alive
Cookie: wp-settings-time-1=1717673282; wordpress_test_cookie=WP%20Cookie%20check; wordpress_logged_in_5c437b347e89969da010fee89e1e0841=admin%7C1717839898%7CuQMh54IlCJNhoDzEHggvTctk7HxgpKRuzSocECLviqN%7Cff109325c191c841ce4f6d74eeff2cff7e6cfbc43a817413db608a53cb325d91
Priority: u=1

--18fc1d05-H--
Message: Warning. String match within "/content-encoding/ /proxy/ /lock-token/ /content-range/ /if/ /x-http-method-override/ /x-http-method/ /x-method-override/" at TX:header_name_920450_x-http-method-override. [file "/opt/crs4/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf"] [line "1193"] [id "920450"] [msg "HTTP header is restricted by policy (/x-http-method-override/)"] [data "Restricted header detected: /x-http-method-override/"] [severity "CRITICAL"] [ver "OWASP_CRS/4.2.0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-protocol"] [tag "paranoia-level/1"] [tag "OWASP_CRS"] [tag "capec/1000/210/272"] [tag "PCI/12.1"]
Message: Access denied with code 403 (phase 2). Operator GE matched 5 at TX:blocking_inbound_anomaly_score. [file "/opt/crs4/rules/REQUEST-949-BLOCKING-EVALUATION.conf"] [line "233"] [id "949110"] [msg "Inbound Anomaly Score Exceeded (Total Score: 5)"] [ver "OWASP_CRS/4.2.0"] [tag "anomaly-evaluation"] [tag "OWASP_CRS"]
Message: Warning. Unconditional match in SecAction. [file "/opt/crs4/rules/RESPONSE-980-CORRELATION.conf"] [line "98"] [id "980170"] [msg "Anomaly Scores: (Inbound Scores: blocking=5, detection=5, per_pl=5-0-0-0, threshold=5) - (Outbound Scores: blocking=0, detection=0, per_pl=0-0-0-0, threshold=4) - (SQLI=0, XSS=0, RFI=0, LFI=0, RCE=0, PHPI=0, HTTP=0, SESS=0, COMBINED_SCORE=5)"] [ver "OWASP_CRS/4.2.0"] [tag "reporting"] [tag "OWASP_CRS"]
Apache-Error: [file "apache2_util.c"] [line 271] [level 3] [client 1XX.YYY.ZZZ.1] ModSecurity: Warning. String match within "/content-encoding/ /proxy/ /lock-token/ /content-range/ /if/ /x-http-method-override/ /x-http-method/ /x-method-override/" at TX:header_name_920450_x-http-method-override. [file "/opt/crs4/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf"] [line "1193"] [id "920450"] [msg "HTTP header is restricted by policy (/x-http-method-override/)"] [data "Restricted header detected: /x-http-method-override/"] [severity "CRITICAL"] [ver "OWASP_CRS/4.2.0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-protocol"] [tag "paranoia-level/1"] [tag "OWASP_CRS"] [tag "capec/1000/210/272"] [tag "PCI/12.1"] [hostname "picard.test.domain.de"] [uri "/index.php"] [unique_id "ZmGdSKiGpWWaBe02xHB0yAAAABg"]
Apache-Error: [file "apache2_util.c"] [line 271] [level 3] [client 1XX.YYY.ZZZ.1] ModSecurity: Access denied with code 403 (phase 2). Operator GE matched 5 at TX:blocking_inbound_anomaly_score. [file "/opt/crs4/rules/REQUEST-949-BLOCKING-EVALUATION.conf"] [line "233"] [id "949110"] [msg "Inbound Anomaly Score Exceeded (Total Score: 5)"] [ver "OWASP_CRS/4.2.0"] [tag "anomaly-evaluation"] [tag "OWASP_CRS"] [hostname "picard.test.domain.de"] [uri "/index.php"] [unique_id "ZmGdSKiGpWWaBe02xHB0yAAAABg"]
Apache-Error: [file "apache2_util.c"] [line 271] [level 3] [client 1XX.YYY.ZZZ.1] ModSecurity: Warning. Unconditional match in SecAction. [file "/opt/crs4/rules/RESPONSE-980-CORRELATION.conf"] [line "98"] [id "980170"] [msg "Anomaly Scores: (Inbound Scores: blocking=5, detection=5, per_pl=5-0-0-0, threshold=5) - (Outbound Scores: blocking=0, detection=0, per_pl=0-0-0-0, threshold=4) - (SQLI=0, XSS=0, RFI=0, LFI=0, RCE=0, PHPI=0, HTTP=0, SESS=0, COMBINED_SCORE=5)"] [ver "OWASP_CRS/4.2.0"] [tag "reporting"] [tag "OWASP_CRS"] [hostname "picard.test.domain.de"] [uri "/index.php"] [unique_id "ZmGdSKiGpWWaBe02xHB0yAAAABg"]
Action: Intercepted (phase 2)
Apache-Handler: proxy:unix:/run/php-fpm/www.sock|fcgi://localhost/
Stopwatch: 1717673288251937 3704 (- - -)
Stopwatch2: 1717673288251937 3704; combined=3375, p1=756, p2=2503, p3=0, p4=0, p5=116, sr=0, sw=0, l=0, gc=0
Producer: ModSecurity for Apache/2.9.6 (http://www.modsecurity.org/); OWASP_CRS/4.2.0.
Server: Apache
Engine-Mode: "ENABLED"

--18fc1d05-Z--

```

**Comment by airween** [comment]
Hi @tux3l,

thanks for report - I'm not sure this is a WP exclusion plugin issue, in fact, it's not even a general CRS FP. I'm afraid this is an expected behavior.

Please check your log, especially this line:

```
Message: Warning. String match within "/content-encoding/ /proxy/ /lock-token/ /content-range/ /if/ /x-http-method-override/ /x-http-method/ /x-method-override/" at TX:header_name_920450_x-http-method-override. [file "/opt/crs4/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf"] [line "1193"] [id "920450"] [msg "HTTP header is restricted by policy (/x-http-method-override/)"] [data "Restricted header detected: /x-http-method-override/"] [severity "CRITICAL"] [ver "OWASP_CRS/4.2.0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-protocol"] [tag "paranoia-level/1"] [tag "OWASP_CRS"] [tag "capec/1000/210/272"] [tag "PCI/12.1"]
```

As you can see, the rule [920450](https://github.com/coreruleset/coreruleset/blob/main/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf#L1140-L1195)  checks the HTTP request headers, and if there is any of one is a listed header in [restricted](https://github.com/coreruleset/coreruleset/blob/main/rules/REQUEST-901-INITIALIZATION.conf#L240) set, then the rule triggers.

And your request contains one of this:

```
X-HTTP-Method-Override: PUT
```

You can reduce this list in your [crs-setup.conf](https://github.com/coreruleset/coreruleset/blob/main/crs-setup.conf.example#L591-L599), but I suggest you to create a runtime exclusion only for the affected URI to do this.

**Comment by azurit** [comment]
@tux3l This should be resolved by rule [9507144](https://github.com/coreruleset/wordpress-rule-exclusions-plugin/blob/544494ef97be40632f092f4b6973f0b3bac2eccd/plugins/wordpress-rule-exclusions-before.conf#L199). Try to recheck if you have installed WordPress rule exclusions plugin correctly.

**Comment by tux3l** [comment]
Hi, thanks for your help/feedback!

after a bit of searching i found my issue - it is like pretty dumb...

I install OWASP and wordpress-rule-exclusion both directly from github to get on all nodes and OS-versions the same rule-versions (instead of different ones from the repositories due to different OS-versions). As i had used an automation-template which only works for OWASP from repository, i added some configuration i needed manually to a new template. As a result it happened to load the wordpress-rule-exclusions before loading OWASP. After fixing the order of when OWASP and wordpress-rule-exclusion are loaded by httpd/apache, it works as expected.

Sorry for the inconvenience and the unneccessary report, hope i get a real problem next time.

Regards tux3l
